src/Compilers/CSharp/Portable/Compiler/MethodBodySynthesizer.Lowered.cs

```diff
@@ -277,4 +285,329 @@ internal static BoundBlock ConstructSingleInvocationMethodBody(SyntheticBoundNod
                         : F.Block(F.Return(invocation));
         }
     }
+
+    internal class SynthesizedAsIntValueMethod : SynthesizedGlobalMethodSymbol
+    {
+        private readonly Emit.PEModuleBuilder _module;
+
+        public SynthesizedAsIntValueMethod(Emit.PEModuleBuilder module, PrivateImplementationDetails pi)
+            : base(module.SourceModule, pi,
+                   module.Compilation.GetSpecialType(SpecialType.System_Boolean),
+                   PrivateImplementationDetails.AsIntValueName)
+        {
+            this._module = module;
+            var compilation = _module.Compilation;
+            this.SetParameters(ImmutableArray.Create<ParameterSymbol>(
+                new SynthesizedParameterSymbol(this, compilation.GetSpecialType(SpecialType.System_Object), 0, RefKind.None, "o"),
+                new SynthesizedParameterSymbol(this, compilation.GetSpecialType(SpecialType.System_Int32), 1, RefKind.Out, "value")
+                ));
+        }
+
+        internal override void GenerateMethodBody(TypeCompilationState compilationState, DiagnosticBag diagnostics)
+        {
+            ///// <summary>
+            ///// If o is of an integral numeric type and contains a value that is in the range of an int,
+            ///// return true and set the value parameter to that value.
+            ///// </summary>
+            //internal static bool AsIntValue(object o, out int value)
+            //{
+            //    value = 0;
+            //    if (o == null)
+            //    {
+            //        return false;
+            //    }
+            //    Type t = o.GetType();
+            //    if (t == typeof(byte)) { value = (byte)o; return true; }
+            //    if (t == typeof(sbyte)) { value = (sbyte)o; return true; }
+            //    if (t == typeof(short)) { value = (short)o; return true; }
+            //    if (t == typeof(ushort)) { value = (ushort)o; return true; }
+            //    if (t == typeof(int)) { value = (int)o; return true; }
+            //    if (t == typeof(long))
```

Comment:
Possible, but that would make the code depend on a number of new things (like the enum GetTypeCode() returns). I'll leave a comment suggesting that possibility.